VmAgent.Core/Interfaces/SystemOperations.cs

```diff
@@ -261,5 +283,45 @@ public void SetUnixFilePermissions(string filePath, int permissions)
             var unixFileInfo = new UnixFileInfo(filePath);
             unixFileInfo.FileAccessPermissions = (FileAccessPermissions)permissions;
         }
+
+        private void SetUnixOwner(string filePath, string owner)
+        {
+            var passwd = Syscall.getpwnam(owner);
+            if(passwd == null)
+            {
+                throw new ArgumentException($"The specified user {owner} could not be retrieved");
+            }
+            var result = Syscall.chown(filePath, passwd.pw_uid, passwd.pw_gid);
+            if(result != 0)
+            {
+                throw new ArgumentException($"Failed to give ownership of {filePath} to {owner}");
+            }
+        }
+
+        public void SetUnixOwnerIfNeeded(string path)
+        {
+            if (_vmConfiguration != null && _vmConfiguration.RunContainersInUsermode)
+            {
+                FileAttributes attr = File.GetAttributes(path);
+                if ((attr & FileAttributes.Directory) == FileAttributes.Directory)
+                {
+                    DirectoryInfo dirInfo = new DirectoryInfo(path);
+                    IEnumerable<FileInfo> files = dirInfo.EnumerateFiles("*", SearchOption.AllDirectories);
+                    IEnumerable<DirectoryInfo> subDirectories = dirInfo.EnumerateDirectories("*", SearchOption.AllDirectories);
+
+                    foreach (FileInfo file in files)
+                    {
+                        SetUnixOwner(file.FullName, "glados");
+                    }
+
+                    foreach (DirectoryInfo directory in subDirectories)
+                    {
+                        SetUnixOwner(directory.FullName, "glados");
+                    }
+                }
+
+                SetUnixOwner(path, "glados");  
```

Comment:
Is this user defined anywhere?

Reply:
It should at least be a const

Reply:
It can be but this is the only place in code it's used, and it's also present on the startup script which is on the other folder, so a constant is a bit superfluous.